HW2/P2/P2.txt

```diff
@@ -0,0 +1,54 @@
+Author: Xingchi Dai Harvard University
+Class: CS205
+Homework 2
+Problem 2: Parallel Vector Updates
+
+Output N = 10
+
+Serial uncorrelated: 0.275285959244 seconds
+Fine grained uncorrelated: 4.32283806801 seconds
+Medium grained uncorrelated: 5.33071494102 seconds
+
+
+Serial correlated: 0.2825050354 seconds
+Fine grained correlated: 4.04863095284 seconds
+Medium grained correlated: 4.04532194138 seconds
+
+
+
+By looking at results, it is not surprising to see that parallel computing(prange) is actually slower than the serial's.
+As the parallel computing introduces more overhead in this case.
+
+If we change the number of N, the performance ourput is shown as follow.
+
+Serial uncorrelated: 0.272067070007 seconds
+Fine grained uncorrelated: 4.33275508881 seconds
+Medium grained uncorrelated: 5.29543185234 seconds when N is equals to 5
```

Comment:
umm. sometimes if we run multiple times, the plot will look better. I only ran once. i will try to do it again.